src/dev-journal.js

```diff
@@ -0,0 +1,84 @@
+const client = require('./client')
+
+
+const isMonday = new Date().getDay() === 1
+const subtractMe = isMonday ? 3 : 1
+const prevWorkDay = new Date(new Date().setDate(new Date().getDate() - subtractMe))
+  .toISOString()
+  .split('T')[0]
+
+// get last-work-day journals
+const getLWDJournals = async (DEV_JOURNAL) => {
+  const prevDayTasks = await client.tasks
+    .getTasksForProject(
+      DEV_JOURNAL,
+      { opt_fields: 'due_on,custom_fields,name,assignee,projects,workspace' },
+    )
+    .then((r) => r.data.filter(({ due_on }) => due_on === prevWorkDay))
+    .catch((e) => console.error(`Failed to fetch prev-work-day tasks: ${e}`))
+
+  const getSubTasks = (task) => client.tasks
+    .getSubtasksForTask(task, { opt_fields: 'gid,name,completed,resource_type' })
+    .then((r) => r.data)
+    .catch((e) => console.error(`Failed to fetch subtasks: ${e}`))
+
+  return Promise.all(prevDayTasks
+    .map(async ({ gid, name, assignee, projects, workspace, custom_fields }) => {
+      const subTasks = await getSubTasks(gid)
+      return {
+        gid,
+        name,
+        assignee,
+        projects: projects[0].gid,
+        workspace: workspace.gid,
+        incompleteSubTasks: subTasks.filter((t) => !t.completed),
+        completedSubTasks: subTasks.filter((t) => t.completed),
+        customField: custom_fields.find((c) => c.name === 'Last Workday').gid,
+      }
+    }))
+}
+
+// format completed tasks into single string
+//    --> TODO: match format with updates
+const formatLWD = (tasks) => {
+  if (tasks.length) {
+    const notes = tasks.map((t) => t.name)
+    return `* ${notes.join('\n* ')}`
+  }
+  return ''
+}
+
+// create new journals
+module.exports.createJournals = async (DEV_JOURNAL) => {
+  try {
+    const prevJournals = await getLWDJournals(DEV_JOURNAL)
+    prevJournals.map(async ({
+      name,
+      assignee,
+      completedSubTasks,
+      incompleteSubTasks,
+      projects,
+      workspace,
+      customField,
+    }) => {
+      // create sub tasks for each journal (mind rate limiting due to tasks loop)
+      const createJournalTasks = (journal, tasks) => tasks
+        .forEach((t) => client.tasks.createSubtaskForTask(journal, { name: t.name }))
+
+      const params = {
+        name,
+        assignee,
+        completed: false,
+        due_on: `${new Date().toISOString().split('T')[0]}`,
+        projects: [projects],
+        workspace,
+        custom_fields: { [customField]: formatLWD(completedSubTasks) },
+      }
+
+      const newJournal = await client.tasks.createTask(params)
+      createJournalTasks(newJournal?.gid, incompleteSubTasks)
```

Comment:
would this need to be awaited, otherwise the workflow cannot be properly safeguarded by the `try/catch` clause?

on top of that, `createJournalTasks` helper can be instead a `map` and fit in a `Promise.all`.

```js
await Promise.all(incompleteSubTasks.map(
  (t) => client.tasks.createSubtaskForTask(newJournal?.gid, { name: t.name })
))
```

except if there's a rate limit concern, in which case this may need to be fired with synchronous logic:

```js
for (const t of incompleteSubTasks) {
  await client.tasks.createSubtaskForTask(newJournal?.gid, { name: t.name })
  // some kind of backoff/sleep to control such as seen in routine.js
}
```

furthermore, if `newJournal.gid` doesn't exist, would it still make sense to even create subtasks for it?

```js
const { gid } = await client.tasks.createTask(params)
if (!gid) {
  // error or don't do anything?
  return
}
// create subtasks
```